### PR TITLE
Add LVL yield source

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/sources/LVLYieldSource.sol [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    uint256 public immutable network;
+
+    ILvlStaking public constant lvlStaking = ILvlStaking(0x08A12FFedf49fa5f149C73B07E31f99249e40869);
+    ILvlRouter public constant lvlRouter = ILvlRouter(0xBD8638C1fF477275E49aaAe3E4691b74AE76BeCd);
+    ILvlPool public constant lvlPool = ILvlPool(0xA5aBFB56a78D2BD4689b25B8A77fd49Bb0675874);
+
+    IERC20 public constant lvlToken = IERC20(0xB64E280e9D1B5DbEc4AcceDb2257A87b400DB149);
+    IERC20 public constant seniorLlpToken = IERC20(0xB5C42F84Ab3f786bCA9761240546AA9cEC1f8821);
+    IERC20 public weth;
+    uint256 public amountDeposited = 0;
```
Comment (ortutay) [inline src/sources/LVLYieldSource.sol]: rm

File: src/sources/LVLYieldSource.sol [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    
+    uint256 public immutable network;
+
+    ILvlStaking public constant lvlStaking = ILvlStaking(0x08A12FFedf49fa5f149C73B07E31f99249e40869);
+    ILvlRouter public constant lvlRouter = ILvlRouter(0xBD8638C1fF477275E49aaAe3E4691b74AE76BeCd);
+    ILvlPool public constant lvlPool = ILvlPool(0xA5aBFB56a78D2BD4689b25B8A77fd49Bb0675874);
+
+    IERC20 public constant lvlToken = IERC20(0xB64E280e9D1B5DbEc4AcceDb2257A87b400DB149);
+    IERC20 public constant seniorLlpToken = IERC20(0xB5C42F84Ab3f786bCA9761240546AA9cEC1f8821);
+    IERC20 public weth;
```
Comment (ortutay) [inline src/sources/LVLYieldSource.sol:34]: can mark this as immutable:

`IERC20 public immutable weth;`

File: src/sources/LVLYieldSource.sol [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    IERC20 public immutable override generatorToken;
+    IERC20 public immutable override yieldToken;
+    
+    uint256 public immutable network;
+
+    ILvlStaking public constant lvlStaking = ILvlStaking(0x08A12FFedf49fa5f149C73B07E31f99249e40869);
+    ILvlRouter public constant lvlRouter = ILvlRouter(0xBD8638C1fF477275E49aaAe3E4691b74AE76BeCd);
+    ILvlPool public constant lvlPool = ILvlPool(0xA5aBFB56a78D2BD4689b25B8A77fd49Bb0675874);
+
+    IERC20 public constant lvlToken = IERC20(0xB64E280e9D1B5DbEc4AcceDb2257A87b400DB149);
```
Comment (ortutay) [inline src/sources/LVLYieldSource.sol]: I think all of these will be different for Arbitrum, so they should follow similar pattern to `weth`

File: src/sources/LVLYieldSource.sol [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import { IYieldSource } from "../interfaces/IYieldSource.sol";
+import { ILvlStaking, UserInfo } from "../interfaces/lvl/ILvlStaking.sol";
+import { ILvlRouter } from "../interfaces/lvl/ILvlRouter.sol";
+import { ILvlPool } from "../interfaces/lvl/ILvlPool.sol";
+
+library LVLConstants {
+    uint256 public constant NETWORK_BNB = 0;
+    uint256 public constant NETWORK_ARBITRUM = 1;
+    
+    uint256 public constant TOKEN_LVL = 0;
```
Comment (ortutay) [inline src/sources/LVLYieldSource.sol]: nit can we make these different from the values above, so there's no chance of mis-use/collision

File: test/LVLYieldSource.t.sol [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import { ILvlRouter } from "../src/interfaces/lvl/ILvlRouter.sol";
+import { ILvlPool } from "../src/interfaces/lvl/ILvlPool.sol";
+
+contract LVLYieldSourceTest is BaseTest {
+
+    function setUp() public {
+        init();
+    }
+
+    /*
```
Comment (ortutay) [inline test/LVLYieldSource.t.sol]: rm or add TODO

**Review by ortutay — APPROVED** [review]

File: src/sources/LVLYieldSource.sol (hunk 2 above)
Comment (ghost) [reply src/sources/LVLYieldSource.sol:34]: this is stupid but it appears you can't if you set it within an `if` statement which is what we're doing

**Review by ortutay — APPROVED** [review]
lgtm

**Pull request closed without merging**